pkg/controller/generic/qtransform/qtransform.go

```diff
@@ -247,6 +252,14 @@ func (ctrl *QController[Input, Output]) reconcileRunning(ctx context.Context, lo
 			return nil
 		}
 
+		if xerrors.TagIs[DestroyOutputTag](err) {
```

Comment:
what about tear down and finalizers on the the output resource? I think we should restructure the flow here (idk how) to move into `reconcileTearingDown` phase... otherwise the finalizers on the input are still set, it looks a bit strange?

Reply:
idk either, let's them hold this, it anyway is only a quality-of-life improvement.

Reply:
can you have another look, did some changes after our discussion.